client/src/game/shapes/shape.ts

```diff
@@ -699,4 +715,38 @@ export abstract class Shape implements IShape {
         }
         return false;
     }
+
+    // DEPENDENT SHAPES
+
+    addDependentShape(shape: IShape): void {
+        if (this.floorId === undefined || this.layerName === undefined) return;
+
+        shape.setLayer(this?.floorId, this?.layerName);
```

Comment:
the `Shape.setLayer` function itself should probably also update its dependents right? (I can't tag the line itself because it's not part of the PR apparently)

Reply:
Yeah, good call.